### PR TITLE
Add office document formats

### PR DESCRIPTION
**Pull request by magbyr**
Adds formats:

-  **doc** - ``application/msword``
-  **docx** - ``application/vnd.openxmlformats-officedocument.wordprocessingml.document``
-  **odt** - ``application/vnd.oasis.opendocument.text``
-  **xls** - ``application/vnd.ms-excel``
-  **xlsx** - ``application/vnd.openxmlformats-officedocument.spreadsheetml.sheet``
-  **ods** - ``application/vnd.oasis.opendocument.spreadsheet``
-  **ppt** - ``application/vnd.ms-powerpoint``
-  **pptx** - ``application/vnd.openxmlformats-officedocument.presentationml.presentation``
-  **odp** - ``application/vnd.oasis.opendocument.presentation``

### CODE REVIEW

**Comment by magbyr** [comment]
Hope these changes can be useful for the project. We're relying on filetype.py usage in our startup, so we're committed to keeping these formats/signatures updated if needed. Copied some signatures / code from the go version, but added some changes because of differences in the files produced by LibreOffice.

**Review by babenek — COMMENTED** [review]
Optimization suggestions.
UnitTests: complex return (...) cannot produce code coverage for analysis. May be if-else does not overhead...

File: tests/test_types.py [hunk 1]
```diff
@@ -65,3 +65,57 @@ def test_guess_zstd(self):
...
+        kind = filetype.guess(FIXTURES + '/sample.pptx')
+        self.assertTrue(kind is not None)
+        self.assertEqual(kind.mime, 'application/vnd.openxmlformats-officedocument.presentationml.presentation')
+        self.assertEqual(kind.extension, 'pptx')
+
+    def test_guess_odp(self):
+        kind = filetype.guess(FIXTURES + '/sample.odp')
+        self.assertTrue(kind is not None)
+        self.assertEqual(kind.mime, 'application/vnd.oasis.opendocument.presentation')
+        self.assertEqual(kind.extension, 'odp')
```
[the remark this answers is not in the document]
Comment (magbyr) [reply tests/test_types.py]: Have added the extra lines at EOF for 3 files.

File: filetype/types/document.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+            and buf[0:8] == b"\xD0\xCF\x11\xE0\xA1\xB1\x1A\xE1"
+            and (
+                (len(buf) > 515 and buf[512:515] == b"\xEC\xA5\xC1\x00")  # MS Office
+                or (
+                    len(buf) > 2142
+                    and b"\x00\x0A\x00\x00\x00MSWordDoc\x00\x10\x00\x00\x00Word.Document.8\x00\xF49\xB2q"
+                    in buf[2075:2142]
+                )  # LibreOffice
+            )
+        )
```
[the remark this answers is not in the document]
Comment (magbyr) [reply filetype/types/document.py]: I have changed the match methods a bit. Probably better for coverage calculation - but slightly slower. What do you think?

File: filetype/types/document.py [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+                )
+                or (buf[512:520] == b"\x09\x08\x10\x00\x00\x06\x05\x00")
+                or (
+                    len(buf) > 2095
+                    and b"\xE2\x00\x00\x00\x5C\x00\x70\x00\x04\x00\x00Calc"
+                    in buf[1568:2095]
+                )
+            )
+        )
+        return header_match and subheader_match
```
Comment (babenek) [inline filetype/types/document.py:185]:
```suggestion
        if len(buf) > 520 and buf[0:8] == b"\xD0\xCF\x11\xE0\xA1\xB1\x1A\xE1":
            if buf[512:516] == b"\xFD\xFF\xFF\xFF" and (buf[518] == 0x00 or buf[518] == 0x02) \
                or buf[512:520] == b"\x09\x08\x10\x00\x00\x06\x05\x00":
                return True
            elif len(buf) > 2095:
                return b"\xE2\x00\x00\x00\x5C\x00\x70\x00\x04\x00\x00Calc" in buf[1568:2095]
        return False
```
Please, check my suggestion. May be spaces missed/extra.

File: filetype/types/document.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (babenek) [reply filetype/types/document.py]: Probably it is so due you used additional assignments. 
I draft my suggestion. Anyway if it so slowly - rollback the changes. 
Additionally you can test with https://github.com/h2non/filetype.py/pull/132 - sometimes it found uncaught exceptions. But you have to change ```-max_len=262``` to 8K.

[the remark this answers is not in the document]
Comment (magbyr) [reply filetype/types/document.py]: Cool. I did the changes and tested with atheris. Found no errors in new code. Atheris found an error in the isobmff class regarding unicode decoding. I think a simple "errors=ignore" on the decode method could be a solution there...

Anyway. Hopefully the latest changes works better with coverage calculation?

[the remark this answers is not in the document]
Comment (babenek) [reply filetype/types/document.py]: Cool, then may be the PR will me merged. Yes, the fix works #131 .

**Pull request merged**